Slack: boost your team's productivity (#1)
LucasOta/posts: dev.to/Slack-boost-your-teams-productivity.md

#Slack: boost your team's productivity

> Let’s suppose the following scenario: you started working in a five person startup, Slack was a bit messy, but it worked for you. After a while the company grows  and now you have 90 employees and the chat is full with random conversations, pet’s pics, and somewhere there is important information.

Reviewer: ```suggestion
Let’s suppose the following scenario: you started working in a five person startup; Slack was a bit messy, but it worked for you. After a while the company grows up, now there are 90 employees and the chat is full with random conversations, pets' pics, the important information is somewhere in there.
```

> Whether you are in a similar situation, or you just have a TOC with management like me, you must fix that, Slack can be fun (and should, we are no robots), but primarily it’s a work tool meant to improve productivity and communication.

Reviewer: ```suggestion
If you are in a similar situation, or you just have an OCD with management like me, you must fix that.
Slack can be funny (and it should, we are no robots), but primarily it’s a work tool meant to improve the productivity and the communication.
```

> In this post I want to propose some tips that I’ve used on different companies and teams that have improved our workflow.

Reviewer: ```suggestion
In this post I want to propose some tips that improved the workflow in the different companies and teams where I've used it.
```

# Table of Content
1. Don’t be afraid of creating channels

> 2. Prioritize information at first glance

Reviewer: Que queres decir aca? "priorizar informacion a primera vista"?

3. An emoji is worth a thousand words

> 4. We are not tailors, but please, let's use more threads

Reviewer: tailors = sastre ?

5. @channel vs @here
6. Custom status
7. Standardize message formats
8. Remove previews
9. One Single Message
10. Avoid Direct Messages when talking about the project

> 11. Adopting one of this practices is a team work

Reviewer: ```suggestion
11. Adopting good practices is a team work
```

12. Conclusion

##Don’t be afraid of creating channels

> In a company there are a lot of different people (thank god!), with different interests, hobbies, tolerance to other human beings, and so on. It’s very probable that not everyone in the company wants to see a picture of your new puppy or your brand new custom mechanical keyboard full of RGB magic for more FPS, but there are for sure a lot of people that would love to see those things. Go and create new channels `#puppies_lovers` `#hardware_fans` `#music` `#coffee` (you can be a little more creative than me with the names).

Reviewer: ```suggestion
In a company there are a lot of different people (thank God!), with different interests, hobbies, different levels of tolerance to other human beings, and so on. It’s very probable that not everyone in the company wants to see a picture of your new puppy or your brand new custom mechanical keyboard full of RGB magic for more FPS, but for sure there's a lot of people that would love to see those things. 
Go and create new channels like `#puppies_lovers` `#hardware_fans` `#music` `#coffee` (you'll probably be a little more creative than me with the names).
```

This way you can create little communities of people with similar interests without disturbing/spamming general channels.

> Another useful proposal I have for you is to have a side channel for your current team. Sometimes there are roles that work on several teams at the same time, so they might not want to scroll over memes or conversations about the Snyderverse to look for useful information (or maybe he does, in that case he can join your side channel).  And as I love consistency, you could use the same notation for every side channel. Example: `#cms_team` (main channel) and `#cms_team_theFunSide` (side channel to make memes competitions and to spoil your teammates).

Reviewer: ```suggestion
Another useful proposal I have for you is to have a side channel for your current team. Sometimes there are roles that work on several teams at the same time, so they might not want to scroll over memes or conversations about the Snyderverse to look for useful information (or maybe they do, and in that case they may join your side channel).
And as I love consistency, I think you could use the same notation for every side channel. Example: `#cms_team` (main channel) and `#cms_team_theFunSide` (side channel to make memes competitions and to spoil your teammates).
```

##Prioritize information at first glance

> Let’s come back to that role that works with different teams, or that developer immersed on [“The Zone”](https://lifehacker.com/what-is-the-zone-anyway-5920484). They might want to see only the important information, and leave the rest for later.

Reviewer: ```suggestion
Let’s come back to that role that works with different teams, or that developer immersed on [“_The Zone_”](https://lifehacker.com/what-is-the-zone-anyway-5920484). They might want to see only the important information and leave the rest for later.
```

> Overcoming this situation is very simple, just use custom emojis at the beginning of the message:

Reviewer: ```suggestion
Overcoming this situation is very simple. Just use custom emojis at the beginning of each message:
```

 - 🟢 Green circle: represents information that won’t change anyone’s responsibilities, priorities or tasks.

> E.g, if you are writing an update on a thread about a task, or if you are going to the bank for a couple of hours.

Reviewer: ```suggestion
E.g, if you are writing an update on a thread about a task, or if you are going out for a couple of hours.
```

>  - ⚠️ Yellow triangle:  you need someone else’s attention, or action, help, etc.

Reviewer: ```suggestion
 - ⚠️ Yellow triangle: you need someone else’s attention, action, help, etc.
```

E.g., a code review, a mail review, a cry for help inside a task thread, a concern about something.
 - 🟥 Red square: needs urgent attention, something must be done immediately. 
E.g, the production server is down, a high priority email from a client, a client meeting being moved or cancelled, or maybe even a PR/mail that needs to be ready ASAP. Whenever you see this, please, go read the message. 

> >Note: it’s important to use those colors to distinguish quickly between them, but it is also necessary to use different shapes for color-blind people.

Reviewer: ```suggestion
>Note: it’s important to use those colors to distinguish quickly between these types of message, but it is also necessary to use different shapes for color-blind people.
```

##An emoji is worth a thousand words
We are seeking productivity increases here, let’s use emojis to achieve this! How?
Scenario: let’s suppose that you are the Scrum Master and you ask your team “Hey guys, do we really need to have today’s meeting with the design team?”, what would you prefer?
_Option A_
![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/5vunokg39mhofvhvl2h4.png)
_Option B_
![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/dktyh8oz854f3umyvj0a.png)

I think the answer is quite obvious, there can also be a discussion on a thread, but having the emojis as the final answer. Also, if there are only :no: emojis, you can use a :cancelled: emoji on the thread to tell everyone that it has been cancelled.

Short answers are very likely to be replaced with an emoji: yes, no, I don’t know (IDK), thanks, solved, merged, done. It might sound quite robotic or impersonal for you, but remember, this is a work tool that seeks for productivity increases, you can still do funny things on the side channel that I proposed earlier.

> ##We are not tailors, but please, let's use more threads 🧵

Reviewer: ```suggestion
##We are not tailors but please, use more threads 🧵
```

> Following the idea of finding relevant information at first glance, imagine a message starts a long conversation and you don't use a tread. If you need to find a message sent before that conversation, you'll have to scroll a lot. Apart from that, the channel would turn into a mess. Wouldn't it be better having a thread that only people involved on that matter will open?

Reviewer: ```suggestion
Following the idea of finding the relevant information at first glance, imagine that a message starts a long conversation and you don't use a tread. If you needed to find a message sent before that conversation, you'd have to scroll a lot. Apart from that, the channel would turn into a mess.
Wouldn't it be better to have a thread only open by the people involved in that particular issue?
```

![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/bg2ahpohluut0fxas5j0.png)

> Besides, it’s not only about having a nice and well organized chat, but also not to disturb team members not involved in the conversation. Just imagine a conversation of 30 messages between 2 persons on the team channel (or even worse, on a general channel), there  will be a lot of people being disturbed  with 30 notifications unnecessarily.

Reviewer: ```suggestion
Besides, it’s not only about having a nice and well organized chat, but also not disturbing team members who are not involved in the conversation. Just imagine a conversation of 30 messages between 2 persons on the team channel (or even worse, on a general channel): there will unnecessarily be a lot of people being disturbed with 30 notifications.
```

##@channel vs @here

> Keeping in mind the idea of not disturbing people when it’s needless, we have to explain the difference between these two tags. It’s very simple, @channel will notify all channel’s members whether they are active or not, @here will notify only the active members. You can work with people in different time zones, so you don’t  want to bother them (or at least you shouldn’t) on their non-working hours, or to disturb the ones on holidays or on a day off, unless it’s critical to let them know. You can learn more about this [here](https://slack.com/intl/en-ar/help/articles/202009646-Notify-a-channel-or-workspace#:~:text=%40everyone%20notifies%20every%20person%20in,they're%20used%20in%20threads).

Reviewer: Creo que ya usaste esa frase o una demasiado parecida mas arriba

"Keeping in mind the idea of not disturbing people when it’s needless"

> Keeping in mind the idea of not disturbing people when it’s needless, we have to explain the difference between these two tags. It’s very simple, @channel will notify all channel’s members whether they are active or not, @here will notify only the active members. You can work with people in different time zones, so you don’t  want to bother them (or at least you shouldn’t) on their non-working hours, or to disturb the ones on holidays or on a day off, unless it’s critical to let them know. You can learn more about this [here](https://slack.com/intl/en-ar/help/articles/202009646-Notify-a-channel-or-workspace#:~:text=%40everyone%20notifies%20every%20person%20in,they're%20used%20in%20threads).

Reviewer: ```suggestion
Keeping in mind the idea of not disturbing people when it’s needless, we have to explain the difference between these two tags.
It’s very simple: @channel will notify all channel’s members, both those who are active and those who are not, while @here will only notify the active members. You can work with people in different time zones, so you don’t want to bother them (or at least you shouldn’t) on their non-working hours, or to disturb the ones on holidays or on a day off, unless it’s critical to let them know.

You can learn more about this [here](https://slack.com/intl/en-ar/help/articles/202009646-Notify-a-channel-or-workspace#:~:text=%40everyone%20notifies%20every%20person%20in,they're%20used%20in%20threads).
```

>Sorry @here I didn't mean to tag you 😬 

##Custom status

> The idea behind this point is not to micromanage your company but to let your colleagues know whether you are available or not. You may need to go to the grocery store, help your kid with something, cook lunch, go to the bathroom (:poop:), etc. If a teammate needs your help when you are active and you don’t answer within 10/30 minutes or more, they might think you are ignoring them, or that they are bothering you,  or they might infer that you are doing something else. Anyways, we can’t know what the others may think, for that reason  with two clicks we can set our custom status as “AFK” (Away From Keyword), “Lunch”,  “Coffee break”. This way you can avoid misunderstandings or bad feelings!

Reviewer: ```suggestion
The idea behind this point is not to micromanage your company but to let your colleagues know whether you are available or not. You may need to go to the grocery store, help your kid with something, cook lunch, go to the bathroom (:poop:), etc. If a teammate needs your help when you are active and you don’t answer within 10/30 minutes or more, they might think you are ignoring them, or that they are bothering you, or they might infer that you are doing something else. Anyways, we can’t know what the others may think and for that reason with only two clicks we can set our custom status as “AFK” (Away From Keyword), “Lunch”,  “Coffee break”.
This way you can avoid misunderstandings or bad feelings!
```

##Standardize message formats

> This can be a little too much for some of you, but remember that we are trying to boost our team performance by adopting some habits on slack. Look at the following example:

Reviewer: ```suggestion
This can be a little too much for some of you, but remember that we are trying to boost our team performance by adopting some habits Look at the following example:
```

![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/pj69588p7dbfdxxyp5a2.png)
![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/bupi02twq2srsxpjugh9.png)

> In the first image some colleagues may ask for the ticket link, or story number. On the other hand, in the second image we already provide information that’s very likely to be useful for our teammates. This way, by investing some clicks and seconds beforehand , we can save time to our partners and get a response quicker.

Reviewer: ```suggestion
In the first image some colleagues may ask for the ticket link or the story number. Oppositely, in the second image we already provide information that’s very likely to be useful for our teammates. This way, by investing some clicks and seconds beforehand, we can save time to our partners and get a quicker response.
```

> This might seem like a waste of time, but I consider that it helps to have a very clean and well organized chat, and also, as I mentioned, save time in the mid term (once everyone gets acquainted with that message format).

Reviewer: ```suggestion
This might seem like a waste of time, but I consider that it helps to have a very clean and well organized chat and also, as I mentioned, save time in the mid term (once everyone gets acquainted with that message format).
```

##Remove previews
Unless you have a Jira bot integration (for example) that adds value to a link preview, I consider that a preview is just a waste of space and it interferes with a clean chat. I’ll leave an example below.
![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/pt2opxa208582o9aza45.png)

> Also, regarding links, by avoiding pasting an entire link on the message will help having a clean message, just add the link to a word that references it, or just simply write  “Link here” and put the link in those words.

Reviewer: ```suggestion
Also regarding links, avoiding pasting an entire link on the message will help you having a clean message. Just add the link to a word that references it or simply write  __“Link here”__ and put the link in those words.
```

> Also, regarding links, by avoiding pasting an entire link on the message will help having a clean message, just add the link to a word that references it, or just simply write  “Link here” and put the link in those words.

Reviewer: No estoy muy seguro del "in those words"

![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/n4t4p8bme4ke8c6ffpw3.png)
![Alt Text](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/hkxvg3vfdz4r4pridozn.png)

##One Single Message

> There are a couple advantages on writing a single message:

Reviewer: ```suggestion
There are a couple advantages in writing a single message:
```

> - Less notifications to your colleagues, not much to explain here, one message is equal to one notification, information spread on 4 separate messages, 4 notifications for each one on the channel.

Reviewer: ```suggestion
- *Less notifications to your colleagues* - not much to explain here, one message is equal to one notification. With the information spread on 4 separate messages, 4 notifications for each one on the channel.
```

Reviewer: aca podrias hacer algun tipo de ""numeracion"". No te digo vinietas, pero si nombrar el tema al principio de cada oracion en negrita. Se me ocurre que queda mejor ordenado y mas amigable

> - Also talking about notifications, if the whole information is reduced to a single message, your partners can read it from the notification pop-up. If you split the information, you are forcing them to open the chat to know what you are talking about.

Reviewer: Por ejemplo sacar el "Also talking about... " y poner directamente alguna palabra importante en bold

> - Organization and readability: if you write down and send messages as you are thinking of an idea, you are very likely to forget something, or maybe that information makes nonsense all spread out. If you take some more seconds and read the message before sending it, it’ll guarantee you that you’re not sending loose thoughts.

Reviewer: ```suggestion
- *Organization and readability*: if you write down and send messages as you think of an idea, you are very likely to forget something, or that information could makes nonsense all spread out. If you take some more seconds and read the message before sending it, that will guarantee you that you’re not sending loose thoughts.
```

> - If you send several messages to express an idea, where should your teammates start the thread in order not to lose relevant information. It might not seem that serious, but imagine someone tags you on a thread created three days ago and you only have partial information on the initial message, you’ll be forced to scroll the chat to find some messages from a couple days ago. Pretty annoying, isn’t it?

Reviewer: ```suggestion
- If you send several messages to express an idea, where should your teammates start the thread in order not to lose relevant information?. It might not seem that serious, but imagine someone tags you on a thread created three days ago and you only have partial information on the initial message. You’ll be forced to scroll the chat to find some messages from a couple days ago. Pretty annoying, isn’t it?
```

##Avoid Direct Messages when talking about the project

> This is a very common practice (or at least it was in the companies I worked on so far), to start a DM conversation between a few members of the team, or just to DM a partner to talk about something you don’t understand about the project or to make a decision.

Reviewer: Either 'the projects I worked on' or 'the companies I worked in'

> Avoiding this practice by starting a conversation on the team channel (you can also tag the people you consider relevant for that matter) bring several advantages:

Reviewer: ```suggestion
Avoiding this practice by starting a conversation on the team channel (you can also tag the people you consider relevant for the matter) brings several advantages:
```

- The team knows what you are dealing with / working on.

> - Maybe people that you thought were not relevant can help you, or to bring an interesting solution to the problem, give their opinion, etc. (You are a team! This is a Key aspect)

Reviewer: ```suggestion
- Maybe people that you thought were not relevant can help you, or bring interesting solutions to the problem, give their opinions, etc. (You are a team! This is a Key aspect)
```

> - Maybe you are working on something that overlaps with a task that a teammate is working on and you didn’t know.

Reviewer: ```suggestion
- Maybe you are working on something that overlaps with a task that a teammate is working on and you didn’t know it.
```

> - Even if you are talking about something very technical that you think is just a matter of the frontend dev team, you might not know the whole team and some  from QA or backend knows about the framework you are working on. DO NOT underestimate anyone on the team.

Reviewer: ```suggestion
- Even if you are talking about something very technical that you think is just a matter of the frontend dev team, you might not know the whole team and some people from QA or backend knows about the framework you are working on. DO NOT underestimate anyone on the team.
```

> ##Adopting one of this practices is a team work

Reviewer: ```suggestion
##Adopting these practices is a team work
```

> Before adopting one of the tips written here in this article on your team or organization, you might want to discuss it with the rest of your team. After that, it will be a process to fully adopt it, some colleagues  can forget about starting a thread, or to use a Yes emoji reaction instead of writing the answer. In those cases you can use the :start_a_thread: emoji, or :use_emojis: , or :thats_a_paddlin: to help your partners to get used to it.

Reviewer: ```suggestion
Before adopting one of the tips written in this article on your team or organization, you might want to discuss it with the rest of your team. After that, it will be a process to fully adopt it: some colleagues will forget to start a thread or to use a 'Yes' emoji reaction instead of writing the answer. In those cases you can use the :start_a_thread: emoji, or :use_emojis: , or :thats_a_paddlin: to help your partners to get used to it.
```

It is useless if only a few people do it.

##Conclusion

> It's possible that some of this tip don't apply to your company, you might need to ignore some of them, change others or simply create new ones to match your use cases.

Reviewer: ```suggestion
It's possible that some of this tips don't apply to your company. You might need to ignore some of them, change others or simply create new ones to match your use cases.
```

> I really want to know what you think, and if you adopt one of the tips I gave you I want to hear what results it provided to your organization.

Reviewer: ```suggestion
I really want to know what you think, and if you adopt one these I want to hear about the results it provided to your organization.
```

Here you have some useful resources:

> - [Slackmojis](https://slackmojis.com/): It's a very long compilation of useful and funny Slack Emojis.

Reviewer: ```suggestion
- [Slackmojis](https://slackmojis.com/): this is a very long compilation of useful and funny Slack Emojis.
```
